Services/UserCourseStatusHistorizing/classes/class.ilUserCourseStatusHistorizingHelper.php

```diff
@@ -230,7 +230,15 @@ public function setIndividualStartAndEnd($user_id, $course_id, &$payload) {
 
 	public static function hasCertificate($user, $course)
 	{
+		require_once('./Services/Certificate/classes/class.ilCertificate.php');
+		require_once(__DIR__ . '/class.ilPrimitiveCertificateAdapter.php');
 		require_once './Modules/Course/classes/class.ilCourseCertificateAdapter.php';
+
+		$cert = new ilCertificate(new ilPrimitiveCertificateAdapter($course));
+		if($cert->readActive() == 0) {
+		  return false;
+		}
+
 		return ilCourseCertificateAdapter::_hasUserCertificate($user, $course)
 		    && gevCourseUtils::getInstance($course)->getParticipationStatusLabelOf($user) == "teilgenommen";
 	}
```

Comment:
The call of ilCourseCertificateAdapter::_hasUserCertificate seems unncecessary, now that we know it will return false true's all the way. You could just combine bothe conditions, i.e. user is successfull and certificate active, into one statement and return it